### DISCUSSION TITLE
There is a serious coding defect.

### DISCUSSION THREAD

**Discussion by Phuxxi** [ryonakano/reco#211] [Q&A] [answered]
Hi all this is a simple problem that involves several parties.

First off Christian Boxdoerfer makes a truly exceptional search program for Linux called Fsearch.

Ryo Nakano   Reco  -  Record talks to remember the contents later

https://github.com/ryonakano/reco

Take photos and record videos with your webcam

https://webcamoid.github.io/

The problem I am now facing is that when I do a search for files - mp3's, pictures, videos, documents etc., using Fsearch, these programs that have NOTHING to do with any thing, have backed up a file NAME list - of everything on my hard drives... like 1.5 million files.

And Fsearch cannot be faulted, but after much looking I can't see any overt ways to cut specific directories from the search.

You see there are legitimate file name pathways, e.g. Home/Desktop/Elvis Movies/ etc. And then there are these files... 

Mostly these two: And there are copies of EVERY file NAME on the system....

/run/user/1000/doc/by-app/com.github.ryonakano.reco/ etc.

/run/user/1000/doc/by-app/io.github.webcamoid.Webcamoid/ etc.

So the programs have coding defects.

Can the people responsible for the apparently copied or replicated coding error please clean them up.

And Christian - if you could include a way to permanently lock out some pathways when doing a search.

Thanks to all.

Shane T. Hanson

![Idi Amin](https://github.com/ryonakano/reco/assets/72900099/0d1a85a2-235d-44dc-a5b9-0f32f8b47906)

**Comment by ryonakano** [comment, marked as answer]
First of all―neither Reco, Webcamoid, nor FSearch do NOT back up your files. And any of these apps do NOT have coding defects about this.

What happens here is exactly AlternativeOstrich7 answered in [this Reddit](https://www.reddit.com/r/linuxquestions/comments/piuzld/comment/hbscrjt/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button). `/run/user/1000/doc/by-app/` is used to export files/directories which user explicitly allows access to Flatpak apps (like Reco or Webcamoid). By exporting, we can prevent Flatpak apps from accessing unnecessary files/directories (which so-called sandboxing).

If you do think it's weird that your files look like backed up in `/run/user/1000/doc/by-app/`, you can wipe them by trying [this](https://github.com/flatpak/flatpak/issues/3660#issuecomment-640262463). However it would also reset settings (like permission of background activity) of any apps installed by Flatpak, which would cause these apps behave unintentionally.
Note, don't try to delete files under `/run/user/1000/doc/by-app/` directly like by `rm` command, it would delete the original files since they're not copy of original, but just exported!

**Reply by Phuxxi** [reply]
You have misread my correspondence. I wrote that the Reco and Webcamonoid are indexing and making file name lists of everything on my hard drive. This is different to making copies of all the files.

All the other Git Hub sourced programs confine their system integration to their own files and their locations within the operating system - they do not perform an indexing of the file system.

Reco and Webcamonoid - do an index of all the files in the system, Fsearch only finds them.... the files and the file name lists as produced by Reco and Webcamonoid.


So yes there is a defect in the way these programs operate - they should not be compiling lists of all the file NAMES in the system....


"which user explicitly allows access to Flatpak apps" - I never gave explicit access to any Flatpack apps to perform these broad ranging system wide access to all the entire directory structure, and the fact that these programs are doing it - shows there is a defect in the programming of them.


They ought not to have full system access and indexation of all the files in the system - only their own directories and the products of their operations in their own directories.

This is what needs to be fixed.


Stay in contact and we can sort it out.


Shane

**Reply by ryonakano** [reply]
> You have misread my correspondence. I wrote that the Reco and Webcamonoid are indexing and making file name lists of everything on my hard drive. This is different to making copies of all the files.

Ah you're right, sorry for misunderstanding your point.

However, I don't agree with your thought that `there is a defect in the way these programs operate`.
This does not due to a defect of Reco or Webcamonoid, but is a feature of Flatpak. This is due to how Flatpak achieves to shut these two apps in a sandboxed environment. This is not what needs to be fixed because it's a feature, not a bug.

> "which user explicitly allows access to Flatpak apps" - I never gave explicit access to any Flatpack apps to perform these broad ranging system wide access to all the entire directory structure, and the fact that these programs are doing it - shows there is a defect in the programming of them.

I'm not sure about Webcamonoid, but Reco does not have permission to access to any of your data. Instead, Reco get permission only to the location where user specified when they save recording or set default saving folder.
Maybe didn't you specified your Desktop folder for default saving location or saved your recording file to Desktop folder at least once? At this point Flatpak "indexes" the file lists under that folder to the /run folder (I'm not good with words, so for more info see [this part of the documentation of Flatpak](https://github.com/flatpak/flatpak/wiki/Portals#the-filechooser-portal:~:text=The%20file%20chooser,to%20the%20app.)):

![スクリーンショット 2023-11-03 23 19 19](https://github.com/ryonakano/reco/assets/26003928/c3e7132e-a5a9-4b4f-8894-cfb091dae11d)

> They ought not to have full system access and indexation of all the files in the system - only their own directories and the products of their operations in their own directories.

Again and conclusion, Reco does not have any file access permission by default already, just get necessary permission only when user allowed, and "indexing" is a feature of Flatpak to manage apps running in a sandboxed environment.

**Reply by Phuxxi** [reply]
Like these programs are running improperly - at some kind of a guess - they should only be operating on 50 - 200 files in their own folder, and not across all the files in the entire hard drive.

Rounding up - that's some 60,000 files have linked to github.ryonakano.reco  AND io.github.webcamoid.Webcamoid.

They should not be running amoke - by building catalogs of file lists.

e.g.

![github ryonakano reco](https://github.com/ryonakano/reco/assets/72900099/76a139b0-b201-4b39-803c-2e32b470ec58)


![github webcamoid Webcamoid](https://github.com/ryonakano/reco/assets/72900099/66bda579-7b1b-4f11-a296-f5d7b9322061)

**Reply by Phuxxi** [reply]
Flat Pack installation offers NO options... so I cannot be saying yes or no to anything....

All the other Flat Pack and or Github sourced programs, none of them do this either.....

There is something very fucking wrong with these

**Reply by ryonakano** [reply]
> All the other Flat Pack and or Github sourced programs, none of them do this either.....

No, this phenomenon would happen all Flatpak apps that can open folders.

For example if you install [FireFox](https://flathub.org/ja/apps/org.mozilla.firefox) and change download folder to your home folder for example, you should get similar result:

```
user@VirtualBox-6b8506d0:~$ find /run/user/1000/doc/by-app/org.mozilla.firefox/d16d3f8e/user/ | wc -l
365839
user@VirtualBox-6b8506d0:~$ 
```

Again, Reco does not involve this behavior since it's a feature of Flatpak, so there are nothing we can take action in the scope of Reco.

**Comment by Me2U2** [comment]
I never said it was backing up anything.... it's making duplicate file lists of EVERYTHING on the system - and it should not be doing this.   You have been a liar from the outset. You have done NOTHING about fixing this problem. Except MORE lying.

**Reply by ryonakano** [reply]
I think this is an expected behavior of Flatpak as I commented before and there is nothing we can take action against this "issue" in the scope of Reco. This is why I closed this discussion.

You should report this to the Flatpak repository if you really think this is a problem.

**Comment by Me2U2** [comment]
Your application and ONE other - are the ONLY ones that access and index the entire filing system....

I raised this with you a long time ago and you just told me to fuck off, deleted my posting and refused to act on the issue.

**Reply by ryonakano** [reply]
> Your application and ONE other - are the ONLY ones that access and index the entire filing system....

I don't think so. As [I tested and commented previously](https://github.com/ryonakano/reco/discussions/211#discussioncomment-7479409), you should face the similar phenomenon with Flatpak FireFox if you select the entire Home folder as a download folder.

The discussion you raised in Reco still exists here and I've just closed it today as out of scope and stale; I don't think this is an "issue" of Reco and there has been no replies since [I commented last time](https://github.com/ryonakano/reco/discussions/211#discussioncomment-7479409).
I'm not refusing to act on your discussion but just confusing to it because I don't feel Reco is related to your "issue"; At least, I haven't written any code to index files under `/run/user/1000/doc/by-app/com.github.ryonakano.reco/` and that would be the job of Flatpak itself. So, you should [report to Flatpak](https://github.com/flatpak/flatpak/issues) and discuss with people who are familiar with Flatpak rather than me if you really think this is a problem.

**Comment by Me2U2** [comment]
Your a fucking liar and you just can't take a HINT or a direct message.
Quit bullshitting me.

**Reply by ryonakano** [reply]
I don't mean to say lies at all. I took time to read your discussion, researched against it, and concluded that it's out of scope of Reco in my opinion, which might be wrong, but not a lie.

BTW, I'm sorry but honestly I'm not sure why you're so insisting on this. You can just quit using Reco if you really think this is an issue of Reco because the maintainer does not seem to willing to "fix" it (in general, a maintainer wouldn't discourage someone to use their app though……).
